pallets/xyk/rpc/src/lib.rs

```diff
@@ -10,11 +10,11 @@ use sp_runtime::{
     traits::{Block as BlockT, MaybeDisplay, MaybeFromStr},
 };
 use std::sync::Arc;
-use xyk_runtime_api::RpcResult;
 pub use xyk_runtime_api::XykApi as XykRuntimeApi;
+use xyk_runtime_api::{RpcAmountsResult, RpcResult};
 
 #[rpc]
-pub trait XykApi<BlockHash, Balance, ResponseType> {
+pub trait XykApi<BlockHash, Balance, AssetId, ResponseType, ResponseTypeTupple> {
```

Comment:
Typo: `Tupple` -> `Tuple`
Also, it would be nice to have descriptive names for response types, like "PriceResponse" or "BurnAmountResponse"